packages/myst-frontmatter/src/licenses/validators.ts

```diff
@@ -103,7 +103,8 @@ export function validateLicense(input: any, opts: ValidationOptions): License |
   if (typeof input === 'string') {
     const value = validateString(input, opts);
     if (value === undefined) return undefined;
-    const valueSpdx = correctLicense(value);
+    // Do not try to coerce long values into SPDX licenses
+    const valueSpdx = value.length < 50 ? correctLicense(value) : undefined;
```

Comment:
Now, we still use the same utility, but we only use it if the provided ID is a reasonable ID length...

Maybe this still isn't the best solution - It would be nice to just have a smarter tool that doesn't just pick up on any substring. But for now, this is an easy improvement.

Reply:
Can we cut it down further to 10 characters? Not sure what the longest id string is, but that probably is the longest we would want to coerce?

Reply:
Yeah, that makes sense. The longest spdx id is 36 characters I think, but that includes a lot of long weird ones.

I might cut it down to 15 - Looking at `CC-BY-NC-ND-4.0` which is a normal, CC license (and 15 chars).

Reply:
```suggestion
    const valueSpdx = value.length < 15 ? correctLicense(value) : undefined;
```